clients/src/main/java/org/apache/kafka/common/requests/TxnOffsetCommitRequest.java

```diff
@@ -70,21 +72,35 @@ public Builder(final String transactionalId,
                        final String memberId,
                        final int generationId,
                        final Optional<String> groupInstanceId) {
+            this(transactionalId, consumerGroupId, producerId, producerEpoch, pendingTxnOffsetCommits, memberId, generationId, groupInstanceId, true);
```

Comment:
I feel like we shouldn't set isTransactionV2Enabled to true by default, can we make it a compulsory argument when we build the request so avoid unexpected behavior especially in tests

Reply:
Sounds good, done.